Source/Fuse.Scripting/Tests/ScriptClass/ScriptStringPromise.Test.uno

```diff
@@ -0,0 +1,46 @@
+using Uno;
+using Uno.Threading;
+using Uno.Testing;
+using Uno.Collections;
+
+using Fuse.Reactive;
+using Fuse.Scripting;
+using Fuse.Controls;
+using FuseTest;
+
+namespace Fuse.Scripting.Test
+{
+	public class CustomStringPanel : Panel
+	{
+		static CustomStringPanel()
+		{
+			ScriptClass.Register(
+				typeof(CustomStringPanel),
+				new ScriptPromise<CustomStringPanel,string,string>("getString", ExecutionThread.MainThread, getString));
+		}
+
+		static Future<string> getString(Context context, CustomStringPanel self, object[] args)
+		{
+			return self.StringFuture;
+		}
+
+		public Future<string> StringFuture { get; set; }
+	}
+
+	public class ScriptStringPromiseTest : TestBase
+	{
+		[Test]
+		public void StringTest()
+		{
+			var child = new UX.StringBasicTest();
+			child.customStringPanel.StringFuture = new Promise<string>("this is a string");
+			using (var root = TestRootPanel.CreateWithChild(child))
+			{
+				while (string.IsNullOrEmpty(child.StringProperties.StringResult))
```

Comment:
I'd like to see a tighter requirement on this. I'm thinking a `MultiStepFrameJS(2)` would be correct here. We expect only the first JS to run and request the future, then the UI thread produce the value, then one more JS step to get the value out.

Reply:
Yeah, this was my crappy attempt at making this more robust, but I think it's been shown that I misidentified the problem.